### ISSUE TITLE
site module has no attribute 'addsitedir'

### ISSUE THREAD

**Issue by flavour** [pyinstaller/pyinstaller#2190]
I am trying to build an .exe for a Web2Py application.
The .exe builds but then crashes at this line:
https://github.com/web2py/web2py/blob/master/gluon/admin.py#L436
AttributeError: 'module' object has no attribute 'addsitedir'
I understand from #768 that PyInstaller has it's own copy of the site module which must be why this happens (I have no other site module around to be loaded in error)

Windows 10, Python 2.7.11, PyInstaller 3.2 & trunk

**Comment by stephenrauch** [comment]
Web2py has its own importer, so it might get tricky to freeze a web2py app.  That said, addsitedir makes no sense in the context of a frozen app, because it is used to help 'find' modules at run time, and the module will need to have been 'found' at compile time.  Often this can be dealt with simply by doing:

```python 

if not getattr(sys, 'frozen', False):
    site.addsitedir('blah blah')

```

But this also means that it may be necessary to make extensive use of hidden imports.

**Comment by htgoebel** [comment]
Changing pyinstaller to detect this import-hack would not help, neither would adding a dummy `addsitedir` function. There is no such thing like a site-dir when running a frozen app (otherwise it would not be frozen but only Parfait :-) Adding a dummy `addsitedir` function would even be worse since it would leave this hack undiscovered.

> Web2py has its own importer, so it might get tricky to freeze a web2py app

So I'm afraid @flavour  needs to get in touch with the web2py developers. Adding to a side-dir burried deep in the application is bad style, very bad style. The web2py developers should fix their application.

**Comment by nicozanf** [comment]
It has been finally fixed on web2py's sources with version 2.18.3 (see [here](https://github.com/web2py/web2py/commit/c1e5571de8cdbc4088a1de47a0e6bc70812176bb) ) 

 ;-)
